### ISSUE TITLE
Externally managed database connections are disposed by ManagedConnection

### ISSUE THREAD

**Issue by ggmueller** [JasperFx/marten#1620]
I am trying to manage some of my connections to the database externally in order to be able to manage the transaction lifecycle.

Basically I do something like this to setup the transaction:
```csharp
var npgsqlConnection = new NpgsqlConnection(ConnectionSource.ConnectionString);
await npgsqlConnection.OpenAsync();
var npgsqlTransaction = await npgsqlConnection.BeginTransactionAsync();
var sessionOptions = SessionOptions.ForTransaction(npgsqlTransaction);
```

Then I use a DocumentSession to update the database.

1. The DocumentSession is disposed, calling ManagedConnection.Dispose
2. This Disposes the underlying Connection, regardless if it was supplied externally or not.

I would suggest, that the ManagedConnection should only dispose the underlying connection, when OwnsConnection is true, changing the Dispose of ManagedConnection to

```csharp
public void Dispose()
{
    if (_ownsConnection)
         _connection?.Dispose();
}
```

I could prepare a PR if you are interested

**Comment by mysticmind** [comment]
@ggmueller `_connection` is of type `TransactionState` and the `Dispose()` already checks for https://github.com/JasperFx/marten/blob/056e1d1aacfe8b9fdd0ef035b4457a0a6d84a3db/src/Marten/Services/TransactionState.cs#L162

Apparently we did fix to add a check for Rollback on 4.x branch but the change isn't there in 3.x. I am suspecting the issue lies with the Rollback code on 3.x. Could you please let us know the Marten version which you are using and also the exact repro steps?

**Comment by ggmueller** [comment]
@mysticmind Sorry, took me a bit to confirm. Yes you are right, this is an issue only in 3.x. After upgrading to 4.0-alpha.4 it works now.

Thx

**Comment by mysticmind** [comment]
@ggmueller thanks for the reply and confirming the issue to be on 3.x.

**Comment by mysticmind** [comment]
Fixed via #1696
